### DISCUSSION TITLE
Run an action on an object creation/update

### DISCUSSION THREAD

**Discussion by Anton-Karpenko** [piccolo-orm/piccolo_admin#178] [Q&A]
Hey, is it possible to run a certain action when a new instance is created via the admin panel?

**Comment by dantownsend** [comment]
Not at the moment - but we could add it quite easily.

Piccolo Admin is built on top of `PiccoloCRUD` from the `piccolo_api` package, which already has support for [hooks](https://piccolo-api.readthedocs.io/en/latest/crud/hooks.html).

We just need to add a `hooks` argument to [TableConfig](https://piccolo-admin.readthedocs.io/en/latest/table_config/index.html).

```python
app = create_admin(
    table=[TableConfig(MyTable, hooks=my_post_save_hook)]
)
```

I'll try and do a PR this afternoon - shouldn't take long.

**Reply by Anton-Karpenko** [reply]
Sounds great, thanks a lot!

**Reply by dantownsend** [reply]
Give 0.24.0 a go - it's on PyPI now. Here are the [docs](https://piccolo-admin.readthedocs.io/en/latest/table_config/index.html#hooks).
